baker/formatting.tsx

```diff
@@ -111,6 +119,44 @@ export const parseKeyValueArgs = (text: string): KeyValueProps => {
     return options
 }
 
+export const formatDataValue = (
+    value: number,
+    variableId: LegacyVariableId,
+    legacyVariableDisplayConfig: LegacyVariableDisplayConfigInterface = {},
+    legacyChartDimension: LegacyChartDimensionInterface | undefined
+) => {
+    if (!legacyChartDimension) return
+    const legacyVariableConfig: LegacyVariablesAndEntityKey = {
+        entityKey: {},
+        variables: {
+            [variableId]: {
+                id: variableId,
+                display: legacyVariableDisplayConfig,
+                values: [value],
+            },
+        },
+    }
+
+    const legacyGrapherConfig = {
+        dimensions: [
+            {
+                ...legacyChartDimension,
+            },
+        ],
+    }
+
+    const { table, dimensions } = legacyToOwidTableAndDimensions(
+        legacyVariableConfig,
+        legacyGrapherConfig
+    )
+
+    const formattedValueWithUnit = table
+        .get(dimensions[0].slug)
+        .formatValueLong(table.rows[0][variableId])
+
+    return formattedValueWithUnit
```

Comment:
Clever way to match the behaviour on the Grapher! But this is definitely much too hacky than it should be. These functions aren't tested for this use case. 

We should probably try to extract some of the behaviour in helper functions. 

Also, this is one of those things that is marked as "legacy" but kind of isn't, because we don't have a definite plan for how it _should_ be.